app/src/main/kotlin/org/kotlinlsp/index/db/File.kt

```diff
@@ -9,26 +9,35 @@ data class File(
     val path: String,
     val packageFqName: String,
     val lastModified: Instant,
-    val modificationStamp: Long
+    val modificationStamp: Long,
+    val indexed: Boolean,
 )
 
 fun File.toDto(): FileDto = FileDto(
     packageFqName = packageFqName,
     lastModified = lastModified.toEpochMilli(),
-    modificationStamp = modificationStamp
+    modificationStamp = modificationStamp,
+    indexed = indexed,
 )
 
 @Serializable
 data class FileDto(
     val packageFqName: String,
     val lastModified: Long,
-    val modificationStamp: Long
+    val modificationStamp: Long,
+    val indexed: Boolean,
 )
 
 fun Database.fileLastModifiedFromPath(path: String): Pair<Instant, Long>? {
     return filesDb.get<FileDto>(path)?.let { Pair(Instant.ofEpochMilli(it.lastModified), it.modificationStamp) }
 }
 
+fun Database.fileLastModifiedAndIndexedFromPath(path: String): Triple<Instant, Long, Boolean>? {
```

Comment:
We can have a `Database.file(path: String): FileRecord` here and remove both methods